gap/attr.gi

```diff
@@ -1687,6 +1687,138 @@ function(D)
   return fail;
 end);
 
+InstallMethod(DigraphCore, "for a digraph",
+[IsDigraph],
+function(digraph)
+  local N, lo, topdown, bottomup, hom, lo_var, image,
+  comps, comp, cores, G, vert_labels, D, in_core, n, m, L;
+  if not IsImmutableDigraph(digraph) then
+    digraph := DigraphImmutableCopy(digraph);
+  fi;
+  N := DigraphNrVertices(digraph);
+  if IsEmptyDigraph(digraph) then
+    if N >= 1 then
+      return [1];
+    else
+      return [];
+    fi;
+  fi;
+  digraph := ReducedDigraph(digraph);  # isolated verts are not in core
+  if DigraphHasLoops(digraph) then
+    return DigraphLoops(digraph){[1]};
+  elif not IsConnectedDigraph(digraph) then
+    comps := DigraphConnectedComponents(digraph).comps;
+    DigraphVertexLabels(digraph);
+    cores := [];
+    for comp in comps do
+      D := InducedSubdigraph(digraph, comp);
+      D := InducedSubdigraph(D,
+           Positions(BlistList(DigraphVertexLabels(D), DigraphCore(D)), true));
+      Add(cores, D);
+    od;
+    L := Length(cores);
+    in_core := List([1 .. L], x -> true);
+    n := 1;
+    while n <= L do
+      if n <> L then
+        m := n + 1;
+      else
+        m := 1;
+      fi;
+      while m <> n and in_core[n] do
+        if in_core[m] and DigraphHomomorphism(cores[m], cores[n]) <> fail then
+          in_core[m] := false;
+        fi;
+        if m < L then
+          m := m + 1;
+        else
+          m := 1;
+        fi;
+      od;
+      n := n + 1;
+    od;
+    cores := ListBlist(cores, in_core);
+    vert_labels := [];
+    for G in cores do
+      vert_labels := Concatenation(vert_labels, DigraphVertexLabels(G));
+    od;
+    return vert_labels;
+  elif IsCompleteDigraph(digraph) then
+    return [1 .. N];
+  elif IsSymmetricDigraph(digraph) and IsBipartiteDigraph(digraph) then
+    return DigraphEdges(digraph)[1];
+  fi;
+
+  if IsDigraphCore(digraph) then
+    return DigraphVertexLabels(digraph);
```

Comment:
```suggestion
    return DigraphVertices(digraph);
```

Reply:
I think it might be better to run `SetDigraphVertexLabels(digraph, DigraphVertices(digraph))` right at the beginning of the function, and then keep the `return DigraphVertexLabels(digraph)` parts. Otherwise, since I run `ReducedDigraph(digraph)` right at the beginning of the function, the way the vertices with no in or out neighbours play in to the vertex labelling are forgotten about. As in,
~~~
gap> D := Digraph([[], [3], [2]]);
<immutable digraph with 3 vertices, 2 edges>
gap> DigraphCore(D);
[ 1, 2 ]
~~~
when I really would want `[2, 3]` returned.

To fix my inconsistency with talking about vertex labelling and numbering, I'll remove the cases where I returned vertices with returning the vertex label. Sorry this has taken me a while, by the way - I have been away for a bit.